### PR TITLE
Introduce schema pruning and remove broken makeExecutableSchema override

### PR DESCRIPTION
**Pull request by tlivings**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,7 +51,8 @@ federation (2 subschema services implemented via `GraphQLComponent` and a vanill
   - `dataSources` - an array of data sources instances to make available on `context.dataSources` .
   - `dataSourceOverrides` - overrides for data sources in the component tree.
   - `federation` - make this component's schema an Apollo Federated schema (default: `false`).
-  - `makeExecutableSchema` - an optional custom implementation of the function which builds the executable schema from the provided type definitions, imports and resolvers. Defaults to the version provided by the [`graphql-tools`](https://www.graphql-tools.com/docs/generate-schema) package. Not used when `federation` is set to `true`.
+  - `pruneSchema` - prune the schema according to [pruneSchema in graphql-tools](https://www.graphql-tools.com/docs/api/modules/utils_src#pruneschema)
```
Comment (roymanas) [inline README.md:54]: As both pruneSchema  and pruneSchemaOptions  are optional in GraphQLComponentOptions, we can mention that in the readme, might be more clear to the consumers of the package.

File: lib/__tests__.js [hunk 2]
```diff
@@ -1151,112 +1150,151 @@ Test('federated schema can include custom directive', (t) => {
...
-    st.plan(2);
-    const customMakeExecutableSchema = sinon.spy();
-    const componentA = new GraphQLComponent({
-      ...baseComponentOptions,
-      makeExecutableSchema: customMakeExecutableSchema,
-      federation: undefined
+  
+    const rootComponent = new GraphQLComponent({
+      imports: [component1, component2],
+      pruneSchema: true
```
Comment (roymanas) [inline lib/__tests__.js:1208]: Probably we can remove the pruneSchema option from this composite component as the test case says: `no prune` to be better aligned with the test case, even though in this case, it has no impact.

**Review by roymanas — COMMENTED** [review]
Looks good, added nit comments please see if makes sense.

File: lib/__tests__.js [hunk 3]
```diff
@@ -1151,112 +1150,151 @@ Test('federated schema can include custom directive', (t) => {
...
     });
-
-    st.doesNotThrow(() => componentA.schema, 'builds schema correctly');
-    st.equal(customMakeExecutableSchema.calledOnce, true, 'calls custom makeExecutableSchema');
+  
+    const { schema } = rootComponent;
+  
+    console.log(graphql.printSchema(schema));
```
Comment (brianjquinn) [inline lib/__tests__.js:1213]: guessing this was accidentally left while debugging?

File: lib/__tests__.js [hunk 4]
```diff
@@ -1151,112 +1150,151 @@ Test('federated schema can include custom directive', (t) => {
...
+      },
+      pruneSchema: false
     });
-    st.doesNotThrow(() => componentA.schema, 'builds schema correctly');
+  
+    const { schema } = component;
+  
+    console.log(graphql.printSchema(schema));
+  
+    t.ok(schema.getType('Property'), 'Property type was pruned');
```
Comment (brianjquinn) [inline lib/__tests__.js:1255]: doesn't this assertion prove that this test is not working as intended? The test is titled: "Unused types not removed if prune is false" yet the `Property` type was removed as per the assertion.

Comment (roymanas) [reply lib/__tests__.js:1255]: +1, we need to change the assertion message to `Property type was not pruned` or something like that as in this case we are setting `pruneSchema: false`

File: lib/__tests__.js (hunk 2 above)
Comment (tlivings) [reply lib/__tests__.js:1208]: I think it just needs to be worded clearly.

File: lib/__tests__.js (hunk 3 above)
Comment (tlivings) [reply lib/__tests__.js:1213]: Right

File: lib/__tests__.js (hunk 4 above)
Comment (tlivings) [reply lib/__tests__.js:1255]: Just a typo

**Pull request merged**